### DISCUSSION TITLE
Why must feats_type be raw for TTS ?

### DISCUSSION THREAD

**Discussion by iamanigeeit** [espnet/espnet#5278] [Q&A]
Hello,

In `tts.sh`, it looks like `feats_type` must be `raw`.
https://github.com/espnet/espnet/blob/ed6ee209c93fe72b95a687272a5faf69639b7b49/egs2/TEMPLATE/tts1/tts.sh#L226-L232

But what is the `else` doing here?
https://github.com/espnet/espnet/blob/ed6ee209c93fe72b95a687272a5faf69639b7b49/espnet2/tts/espnet_model.py#L93-L99

And it looks like `write_collected_feats=True` is wasted, because we are going to re-extract the features on the fly during training:
https://github.com/espnet/espnet/blob/ed6ee209c93fe72b95a687272a5faf69639b7b49/egs2/TEMPLATE/tts1/tts.sh#L651-L654

I much prefer extracting all the features first before i train, as it saves time if i want to repeat training or mix different corpora.

**Comment by kan-bayashi** [comment]
This is due historical issue.
1. `tts.sh` and model design are derived from ASR.
2. At first, `asr.sh` want to use kaldi-based acoustic features including pitch since its performance is better.
3. But TTS case, we do not need to use kaldi-based feature, so I omit them to reduce the range of testing.

I understand preference, so if we can improve the design of `tts.sh` to use pre-calculated features, that is perfect.
